### PR TITLE
Test fixtures 413

### LINKED ISSUE #413: Migrate test setup/teardown to fixtures
Right now, the tests are somewhat stateful: Tests modify inputs/outputs of other tests and/or don't clean up after themselves properly. This can cause problems when JSON-Fortran is added as an [`ExternalProject`] in a CMake "SuperBuild" with testing enabled.

What happens is that subsequent runs of the build will see that JSON-Fortran is up to date, skipping the configure and build steps, but will run the test step. (Or, perhaps it will re-run the configure step I'm not 100% clear on this, but I think my initial understanding is correct.) Because the test inputs/outputs are stateful, the initial run after configuring and building will pass, but subsequent runs will fail.

To resolve this the following steps should be taken:

1. Analyze inter-test run order dependencies. CTest allows tests to be run in parallel, and some tests may be taking the output of others as inputs. This should be explicitly declared.
2. CMake/CTest has added support for ["test fixtures"] which allow setup/teardown tasks for groups of tests to be declared. Use this to copy inputs (and cleanup outputs) between test runs.

[`ExternalProject`]: https://cmake.org/cmake/help/v3.14/module/ExternalProject.html
["test fixtures"]: https://cmake.org/cmake/help/v3.14/prop_test/FIXTURES_REQUIRED.html

### PR DESCRIPTION
**Pull request by zbeekman**
Fixes #413 

Also, re-enables & fixes some validation of outputs created by unit tests.

### CODE REVIEW

**Comment by zbeekman** [comment]
Also, removes reliance on diff to check output for regressions.

**Comment by zbeekman** [comment]
Assuming no test failures (or only due to #422) this PR should be ready to merge. If you have any questions, or need more info, LMK.

**Comment by zbeekman** [comment]
@jacobwilliams let me know if there's anything you want me to explain or cleanup. Otherwise, I believe that this is ready to merge.

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,6 +14,9 @@ cmake_minimum_required ( VERSION 2.8.8 FATAL_ERROR )
 # Use MSVS folders to organize projects on windows
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 
+option(JSON_FORTRAN_USE_OpenCoarrays
+  "Build VTKmofo with support for linking against OpenCoarray programs" OFF)
```
Comment (jacobwilliams) [inline CMakeLists.txt]: VTKmofo?

**Comment by jacobwilliams** [comment]
Do we need the OpenCoarrays stuff in the Cmake file? If it's useful we can keep it, can you provide me with some text I can put in the release notes explaining what that is for?

**Comment by jacobwilliams** [comment]
Can you pull latest master to get the change from #422?

File: CMakeLists.txt (hunk 1 above)
Comment (zbeekman) [reply CMakeLists.txt]: Derp.... I saw that and meant to fix it but then lost track of it. (https://github.com/porteri/vtkmofo)

File: CHANGELOG.md [hunk 2]
```diff
@@ -38,6 +38,15 @@
...
 [Complete Changeset](https://github.com/jacobwilliams/json-fortran/compare/7.1.0...HEAD)
 
+**Enhancements:**
+
+- Allow linking to JSON-Fortran from GFortran programs using
+  [OpenCoarrays](https://github.com/sourceryinstitute/OpenCoarrays#readme)
+  as the coarray runtime implementation. Use the
+  `-DJSON_FORTRAN_USE_OpenCoarrays:BOOL=ON` option to cmake to enable
+  this. (NOTE: The fact that this is required may be a bug in
+  GFortran.)
```
Comment (zbeekman) [inline CHANGELOG.md]: @jacobwilliams how's this for the explanation? If you prefer, I can remove this option and users can manually tweak CMake flags if they need to.

Comment (jacobwilliams) [reply CHANGELOG.md]: That sounds fine. I don't mind keeping it if it's useful.

**Pull request merged**